### ISSUE TITLE
Parse error on dashboard variable query

### ISSUE THREAD

**Issue by nothub** [grafana/grafana#53248]
**What happened**:
When trying to create a dashboard variable, the parser failed to parse the following query: `(count(count by(cpu) (node_cpu_seconds_total{job="qnap-node"})))`

**What you expected to happen**: The query to be correctly parsed and executed

**How to reproduce it (as minimally and precisely as possible)**:

![image](https://user-images.githubusercontent.com/48992448/182642862-2510d75d-4370-43c2-81f0-672e6e209f81.png)

**Environment**:
- Grafana version: 9.0.2

**Comment by zuchka** [comment]
I can confirm this error throws on 9.0.5 but ONLY in the Prometheus Query-based variable view. That PromQL command works from explore and the panel's query editor. Here is a screenshot of the UI and the output from G-server on debug-level:

```
ts.kvstore t=2022-08-04T00:18:48.409234048Z level=debug msg="got secret value" orgId=1 type=datasource namespace=Prometheus
logger=tsdb.prometheus t=2022-08-04T00:18:48.409772132Z level=debug msg="Sending resource query" URL="api/v1/series?match%5B%5D=(count(count%20by(cpu)%20(node_cpu_seconds_total%7Bjob%3D%22qnap-node%22%7D)))&start=1659550728&end=1659572328"
logger=context traceID=00000000000000000000000000000000 userId=1 orgId=1 uname=admin t=2022-08-04T00:18:48.882664507Z level=info msg="Request Completed" method=GET path=/api/datasources/1/resources/api/v1/series status=400 remote_addr=172.17.0.1 time_ms=476 duration=476.988042ms size=117 referer="http://localhost:3000/dashboard/new?orgId=1&editPanel=2&editview=templating" traceID=00000000000000000000000000000000
```

![CleanShot 2022-08-03 at 14 20 52@2x](https://user-images.githubusercontent.com/37156449/182737224-ac71068d-d921-48e0-9fd0-2cf571840e25.png)

**Comment by evandandrea** [comment]
This may intersect with the streaming parser work that @itsmylife is driving. Worth checking in with them.

**Comment by gtk-grafana** [comment]
So the backend is simply proxying the request directly to prometheus, so a local url that exhibits the error hits the grafana backend:
`http://localhost:3000/api/datasources/5/resources/api/v1/series?match%5B%5D=count(process_cpu_seconds_total)&start=1660656584&end=1660678184`
from
`/grafana/grafana/pkg/tsdb/prometheus/client/client.go::QueryResource`

Returns the following url: `http://localhost:9090/api/v1/series?match%5B%5D=count(process_cpu_seconds_total)&start=1660656584&end=1660678184`.
Which returns the error we're seeing on the FE:
```{"status":"error","errorType":"bad_data","error":"invalid parameter \"match[]\": 1:6: parse error: unexpected \"(\""}```

I'm still looking into things, but that looks as though the error is coming directly from prometheus, so I'm going to spend some time with their documentation and see if I can learn more about this API endpoint and its usage.

**Comment by gtk-grafana** [comment]
So the prometheus [series endpoint](https://prometheus.io/docs/prometheus/latest/querying/api/#finding-series-by-label-matchers), only accepts an array of [series selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors) strings to match against existing series in the database, this endpoint isn't evaluating any arbitrary prometheus query and subsequently using the result of that execution to create values for a "variable" in grafana, it's simply a reference to an existing time series, so I would propose that this is not a bug, and the application is behaving as it should (although it wouldn't hurt to update the documentation so the expectation of the variable field is set more appropriately).

TL;DR Variables are used to allow users to quickly toggle between specific time series and must be a reference to (one or more) existing time series.

@nothub I would love to hear more about your use case that would require this functionality, I wonder if there's another way we can set up your dashboard to get you the desired graphs without running a function in a variable?

**Comment by nothub** [comment]
My use case for this is just a small one. In a dashboard that is monitoring a single target, I was trying to enhance the queries readability by following the DRY principle and moving the count of cores to a variable.

**Comment by gtk-grafana** [comment]
@grafana/observability-metrics Needs to re-investigate as this might have been solved in another PR.

**Comment by bohandley** [comment]
Investigating to see if this PR solved this issue https://github.com/grafana/grafana/pull/51411

**Comment by bohandley** [comment]
Hi @nothub, after investigating the above PR, it looks like that was unrelated.

I have a question though, for a Prometheus template variable, to return results for a query, one must wrap the query in the `query_result()` function like so:

`query_result((count(count by(cpu) (node_cpu_seconds_total{job="qnap-node"}))))`

It looks like you are running the query directly. Does passing the query to the `query_result()` function work for you?

I was able to run the following query successfully:

`query_result((count(count by(cpu) (go_gc_duration_seconds{job="prometheus"}))))`

**Comment by nothub** [comment]
Thanks a lot @bohandley :+1: !
The `query_result()` function produces a list containing the required value and with a regex like `/\s(.+)\s/` i can reduce the list to only the value.
